core/lib/src/response/responder.rs

```diff
@@ -217,34 +219,156 @@ impl<'r> Responder<'r> for String {
     }
 }
 
+fn range_response<'r, B: io::Read + io::Seek + 'r>(range: Range, mut body: B) -> Result<response::Response<'r>, B> {
+    use std::cmp;
+    use http::hyper::header::{ContentRange, ByteRangeSpec, ContentRangeSpec};
+
+    match range {
+        Range::Bytes(ranges) => {
+            if ranges.len() == 1 {
+                let (start, end, total) = match ranges[0] {
+                    ByteRangeSpec::FromTo(mut start, mut end) => {
+                        if end < start {
+                            return Ok(
+                                Response::build()
+                                    .status(Status::RangeNotSatisfiable)
+                                    .header(AcceptRanges(vec![RangeUnit::Bytes]))
+                                    .finalize()
+                            );
+                        }
+                        let size = body.seek(io::SeekFrom::End(0))
+                            .expect("Attempted to retrieve size by seeking, but failed.");
+                        if start > size {
+                            start = size;
+                        }
+                        if end > size {
+                            end = size;
+                        }
+                        body.seek(io::SeekFrom::Start(start))
+                            .expect("Attempted to seek to the start of the requested range, but failed.");
+                        (start, end, size)
+                    },
+                    ByteRangeSpec::AllFrom(mut start) => {
+                        let size = body.seek(io::SeekFrom::End(0))
+                            .expect("Attempted to retrieve size by seeking, but failed.");
+                        if start > size {
+                            start = size;
+                        }
+                        body.seek(io::SeekFrom::Start(start))
+                            .expect("Attempted to seek to the start of the requested range, but failed.");
+                        (start, size, size)
```

Comment:
Should this be `(start, size - start, size)` ?

Reply:
fixed